lib/tasks/multitenant_db_merge.rake

```diff
@@ -0,0 +1,344 @@
+  }
+  @fulfillment_mappings = easy_mass_insert Fulfillment, 'fulfillments', fulfillment_map, false
+
+  # Port practical supports
+  psup_map = -> (x) {
+    res = x.except('id', 'created_at', 'updated_at', 'can_support_id', 'fund_id')
+      .merge({
+        'fund_id' => @fund_id,
+        'created_at' => x['created_at'].asctime.in_time_zone("America/New_York"),
+        'updated_at' => x['updated_at'].asctime.in_time_zone("America/New_York"),
+        'can_support_id' => if x['can_support_type'] == 'Patient'
+                           @patient_mappings[x['can_support_id']]
+                         elsif x['can_support_type'] == 'ArchivedPatient'
+                           @archived_patient_mappings[x['can_support_id']]
+                         else
+                           raise "unexpected type - row #{x}"
+                         end
+      })
+    res['can_support_id'].nil? ? nil : res
+  }
+  @practical_support_mappings = easy_mass_insert PracticalSupport, 'practical_supports', psup_map, false
+
+  # Port notes
+  note_map = -> (x) {
+    res = x.except('id', 'created_at', 'updated_at', 'fund_id', 'patient_id')
+           .merge({
+             'fund_id' => @fund_id,
+             'created_at' => x['created_at'].asctime.in_time_zone("America/New_York"),
+             'updated_at' => x['updated_at'].asctime.in_time_zone("America/New_York"),
+             'patient_id' => @patient_mappings[x['patient_id']]
+           })
+    res['patient_id'].nil? ? nil : res
+  }
+  @note_mappings = easy_mass_insert Note, 'notes', note_map, false
+
+  # Port call list entries
+  cle_map = -> (x) {
+    res = x.except('id', 'created_at', 'updated_at', 'fund_id', 'line_id', 'patient_id', 'user_id')
+           .merge({
+             'fund_id' => @fund_id,
+             'created_at' => x['created_at'].asctime.in_time_zone("America/New_York"),
+             'updated_at' => x['updated_at'].asctime.in_time_zone("America/New_York"),
+             'line_id' => @line_mappings[x['line_id']],
+             'patient_id' => @patient_mappings[x['patient_id']],
+             'user_id' => @user_mappings[x['user_id']]
+           })
+    res['patient_id'].nil? ? nil : res
+  }
+  @call_list_entry_mappings = easy_mass_insert CallListEntry, 'call_list_entries', cle_map, false
+
+  # Port events
+  event_map = -> (x) {
+    res = x.except('id', 'created_at', 'updated_at', 'fund_id', 'line_id', 'patient_id')
+           .merge({
+             'fund_id' => @fund_id,
+             'created_at' => x['created_at'].asctime.in_time_zone("America/New_York"),
+             'updated_at' => x['updated_at'].asctime.in_time_zone("America/New_York"),
+             'line_id' => @line_mappings[x['line_id']],
+             'patient_id' => @patient_mappings[x['patient_id'].to_i],
+           })
+    res['patient_id'].nil? ? nil : res
+  }
+  @event_mappings = easy_mass_insert Event, 'events', event_map, false
+
+  # Versions, maybe a little more complicated...
```

Comment:
I'm gunna need a bit more verbosity from you here friend 😂 Can you give me a bit more on how you intend the mappings and the helper function feed into the versions port lambda?

Reply:
lol I took a whack, hopefully that's helpful